packages/aws-cdk/test/api/sdk-provider.test.ts

```diff
@@ -138,6 +149,40 @@ describe('CLI compatible credentials loading', () => {
 
     await expect(provider.forEnvironment(`${uid}some_account_#`, 'def', Mode.ForReading)).rejects.toThrow('Need to perform AWS calls');
   });
+
+  test('even when using a profile to assume another profile, STS calls goes through the proxy', async () => {
+    // Messy mocking
+    let called = false;
+    jest.mock('proxy-agent', () => {
+      class FakeAgent extends require('https').Agent {
```

Comment:
does this not complain about using require?

Reply:
Of course it does :)